src/main/java/frc/robot/commands/States/prep_coral/PrepCoralLv.java

```diff
@@ -5,51 +5,184 @@
 package frc.robot.commands.States.prep_coral;
 
 import frc.robot.subsystems.StateMachine.RobotState;
+
+import edu.wpi.first.math.geometry.Pose2d;
+
+import edu.wpi.first.units.Units;
 import edu.wpi.first.units.measure.Distance;
 import edu.wpi.first.wpilibj2.command.Command;
 import frc.robot.subsystems.Motion;
 import frc.robot.subsystems.Rotors;
+import frc.robot.subsystems.Drivetrain;
 import frc.robot.subsystems.StateMachine;
+import frc.robot.Constants.MechanismPositionGroup;
+import frc.robot.Constants.constField;
 import frc.robot.Constants.constMechanismPositions;
+import edu.wpi.first.units.measure.Angle;
 
 /* You should consider using the more terse Command factories API instead https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#defining-commands */
 public class PrepCoralLv extends Command {
   /** Creates a new PrepCoralLv. */
+  Drivetrain globalDrivetrain;
   Motion globalMotion;
   Rotors globalRotors;
   StateMachine globalStateMachine;
-  Distance globalDistance;
+  Distance globalHeight;
+  Angle drivetrainRotation;
+  Pose2d closestPoseByRotation;
+  MechanismPositionGroup prepL2;
+  MechanismPositionGroup prepL3;
+  MechanismPositionGroup prepL4;
+  Distance backReefDistance;
+  Distance frontReefDistance;
 
-  public PrepCoralLv(StateMachine globalStateMachine, Motion subMotion, Rotors subRotors, Distance height) {
+  public PrepCoralLv(StateMachine globalStateMachine, Motion subMotion, Rotors subRotors, Distance height,
+      Drivetrain subDrivetrain) {
     // Use addRequirements() here to declare subsystem dependencies.
```

Comment:
I suggest putting subDrivetrain before the height so it goes with the other subsystem (will still work as is but a little less clean)